### PR TITLE
feat(ingest) - Option to ingest custom properties as patch

### PR DESCRIPTION
**Pull request by treff7es**
## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/source/s3/config.py [hunk 1]
```diff
@@ -70,6 +70,12 @@ class DataLakeSourceConfig(PlatformSourceConfigBase, EnvBasedSourceConfigBase):
         "path_spec", "path_specs", lambda path_spec: [path_spec]
     )
 
+    ingest_patches: bool = Field(
```
Comment (shirshanka) [inline metadata-ingestion/src/datahub/ingestion/source/s3/config.py:73]: Suggest having per-aspect knobs for controlling patch-based application or not versus a global one. cc @hsheth2 for your thoughts

Comment (shirshanka) [reply metadata-ingestion/src/datahub/ingestion/source/s3/config.py:73]: e.g. 
`patch_properties: true`
`patch_tags: true`

Comment (hsheth2) [reply metadata-ingestion/src/datahub/ingestion/source/s3/config.py:73]: Agree that per-aspect knobs is better

**Comment by jjoyce0510** [comment]
We will reopen this later. 

Cheers
John

**Pull request closed without merging**